src/brevitas/graph/calibrate.py

```diff
@@ -195,41 +216,46 @@ def correct_bias_hook(self, module, inp, name, parent_module):
             quant_mean = self.compute_mean(inp, transpose_dim)
             error = self.float_mean_map[name] - quant_mean
             self.update_correction(name, error)
-            if self.curr_iter == self.iterations - 1:
-                self.apply_correction(name, parent_module)
             del self.float_mean_map[name]
             inp_broadcast_shape = compute_channel_view_shape(inp, channel_dim=self.channel_dim(inp, parent_module))
             return inp + error.reshape(inp_broadcast_shape)
 
-    def collect_float_mean(self, model, *args, **kwargs):
+    def register_collect_float_mean_hook(self, model):
         for name, module in model.named_modules():
             if isinstance(module, self.layers):
                 hook_fn = partial(self.collect_float_mean_hook, name=name, parent_module=module)
                 hook = module.output_quant.register_forward_pre_hook(hook_fn)
                 self.collect_float_mean_hooks.append(hook)
-        model(*args, **kwargs)
-        for hook in self.collect_float_mean_hooks:
-            hook.remove()
-        self.collect_float_mean_hooks = []
 
-    def correct_bias(self, model, *args, **kwargs):
+    def register_correct_bias_hook(self, model):
         for name, module in model.named_modules():
             if isinstance(module, self.layers):
                 hook_fn = partial(self.correct_bias_hook, name=name, parent_module=module)
                 hook = module.output_quant.register_forward_pre_hook(hook_fn)
                 self.correct_bias_hooks.append(hook)
-        model(*args, **kwargs)
+
+    def float_mean_hooks_cleanup(self):
+        for hook in self.collect_float_mean_hooks:
+            hook.remove()
+        self.collect_float_mean_hooks = []
+
+    def correct_bias_hooks_cleanup(self):
         for hook in self.correct_bias_hooks:
             hook.remove()
         self.correct_bias_hooks = []
-        assert not self.float_mean_map
 
-    def apply(self, model, *model_args, **model_kwargs):
+    def global_forward_pre_hook(self, model, inp):
```

Comment:
It's not obvious to me how this works anymore. Is the implementation tied to bias_correction_mode? We should add a few more comments and potentially rename the class to _BiasCorrection if it's only for internal use